### PR TITLE
Fix eviction occupancy stats decrement

### PR DESCRIPTION
**Pull request by arutk**
Eviction should decrement occupancy statistics for the
core from which a cacheline is being evicted rather than
from the I/O target core.

### CODE REVIEW

**Review by mmichal10 — APPROVED** [review]

File: tests/functional/tests/eviction/test_eviction.py [hunk 1]
```diff
@@ -20,17 +20,43 @@
...
-    """
+    """Test if eviction does not occur when IO greater than cache size is submitted."""
     cache_device = Volume(Size.from_MiB(20))
 
     core_device = Volume(Size.from_MiB(5))
-    cache = Cache.start_on_device(cache_device, cache_mode=mode,
-                                  cache_line_size=cls)
-    cache_size = cache.get_stats()['conf']['size']
+    cache = Cache.start_on_device(cache_device, cache_mode=mode, cache_line_size=cls)
+    cache_size = cache.get_stats()["conf"]["size"]
```
Comment (arutk) [inline tests/functional/tests/eviction/test_eviction.py:59]: all the changes below are from the black formatting tool

File: tests/functional/tests/eviction/test_eviction.py [hunk 2]
```diff
@@ -20,17 +20,43 @@
 logger = logging.getLogger(__name__)
 
 
+@pytest.mark.parametrize("cls", CacheLineSize)
+@pytest.mark.parametrize("mode", [CacheMode.WT])
+def test_two_cores(pyocf_ctx, mode: CacheMode, cls: CacheLineSize):
+    """Test if eviction does not occur when IO greater than cache size is submitted."""
```
Comment (rafalste) [inline tests/functional/tests/eviction/test_eviction.py:26]: Based on the test below, shouldn't this one be called something like `test_write_size_greater_than_cache_with_two_cores`? And also update the description?

**Review by rafalste — APPROVED** [review]

File: tests/functional/tests/eviction/test_eviction.py (hunk 2 above)
Comment (arutk) [reply tests/functional/tests/eviction/test_eviction.py]: right, fixe

**Review by rafalste — APPROVED** [review]

**Comment by robertbaldyga** [comment]
retest this please

**Review by rafalste — APPROVED** [review]

**Review by mmichal10 — APPROVED** [review]

**Review by robertbaldyga — APPROVED** [review]

**Pull request merged**